src/System.Management.Automation/engine/CommandDiscovery.cs

```diff
@@ -1131,11 +1131,27 @@ private static CommandInfo InvokeCommandNotFoundHandler(string commandName, Exec
                                 }
 
                                 result = LookupCommandInfo(commandName, commandTypes, searchResolutionOptions, commandOrigin, context);
+
+                                if (result != null)
+                                {
+                                    break;
+                                }
                             }
 
-                            if (result != null)
+                            if (result == null && searchResolutionOptions.HasFlag(SearchResolutionOptions.UseAbbreviationExpansion))
                             {
-                                break;
+                                foreach (var cmdlet in exportedCommands)
+                                {
+                                    string abbreviatedCmdlet = new string(cmdlet.Key.Where(c => char.IsUpper(c) || c == '-').ToArray());
+                                    if (commandName.Equals(abbreviatedCmdlet, StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        result = LookupCommandInfo(cmdlet.Key, commandOrigin, context);
```

Comment:
LookupCommandInfo() does automoduleloading (line 828 in commanddiscovery.cs)

Reply:
`TryModuleAutoLoading` seems only loading the module if the command name is fully qualified. `LookupCommandInfo` depends on `TryModuleAutoDiscovery` to load the module if the auto-discovery finds anything, and that is supposed to happen in `if (exportedCommands.TryGetValue(commandName, out exportedCommandTypes))` via the call to `AutoloadSpecifiedModule`.

Your current implementation actually does 2-pass when the command is discovered from a module:
- in the first pass, you only resolve the abbreviated command name to the full command name, and then call `LookupCommandInfo` with the resolved command name, which will start the 2nd auto-discovery pass because the module is not yet loaded.
- in the second pass, the execution will get in `if (exportedCommands.TryGetValue(commandName, out exportedCommandTypes))`, and the module is loaded via `AutoloadSpecifiedModule`.

Reply:
I believe this is addressed with the additional code change and test added for auto-loading